src/components/landing-page/u30/product-chooser.tsx

```diff
@@ -0,0 +1,88 @@
+"use client";
+
+import { u30Coupons } from "@/components/landing-page/u30/coupons";
+import { Button } from "@/components/ui/button";
+import { SelectField } from "@/components/ui/form";
+import { LandingPageOption } from "@/components/landing-page/product-option";
+import type { AnalyticsObject } from "@/lib/analytics";
+import { css } from "@/theme/css";
+import { useTranslations } from "next-intl";
+
+export function U30Chooser({
+  analyticsParams,
+}: {
+  // TODO remove this again when we don't redirect to legacy /angebote
+  analyticsParams?: AnalyticsObject;
+}) {
+  const t = useTranslations("landing.u30");
+
+  return (
+    <form
+      action={`/angebot`}
+      method="GET"
+      className={css({
+        display: "flex",
+        flexDirection: "column",
+        gap: "8",
+        width: "full",
+      })}
+    >
+      {
+        // TODO remove this again when we don't redirect to legacy /angebote
+        analyticsParams
+          ? Object.entries(analyticsParams).map(([k, v]) => {
+              return <input key={k} name={k} value={v} hidden readOnly />;
+            })
+          : null
+      }
+      <div
+        className={css({
+          display: "flex",
+          flexDirection: "column",
+          gap: "2",
+        })}
+      >
+        <LandingPageOption name="product" value="YEARLY" defaultChecked>
+          <strong>{t("options.yearly")}</strong> für CHF 99 pro Jahr
+        </LandingPageOption>
+
+        <LandingPageOption name="product" value="MONTHLY">
+          <strong>{t("options.monthly")}</strong> für CHF 9 pro Monat
+        </LandingPageOption>
+      </div>
+
+      <div
+        className={css({
+          display: "flex",
+          flexDirection: "column",
+          gap: "2",
+        })}
+      >
+        <p className={css({ fontSize: "lg" })}>{t("tellUsYourAge")}</p>
+
+        <SelectField
+          name="promo_code"
+          label={t("birthYear")}
+          defaultValue={""}
+          hideLabel
+          required
+        >
+          <option value="" selected disabled></option>
+          {u30Coupons.map(({ durationInMonths, promoCode }) => {
+            // get birth year relative to current year and coupon duration
+            const currentYear = new Date().getFullYear();
+            const birthYearLabel = currentYear - 31 + durationInMonths / 12;
+
+            return (
+              <option value={promoCode} key={promoCode}>
+                {birthYearLabel}
+              </option>
+            );
+          })}
+        </SelectField>
```

Comment:
we discussed it already, but i'm not super keen on the dropdown. it feels like we just wanted to get done quickly with the implementation. i would find a text field with some validation/warnings **much** nicer for the user, despite the complexity it entails.
other question: what happens if a 16 years old wants to buy an abo? (with a text field we could simply refer them to support, as this is quite an edge case. still generally speaking, i'm not sure what to do with this case…)

Reply:
Yes, I'll change it